client/src/components/restaurant/restaurant-register.js

```diff
@@ -0,0 +1,224 @@
+    setTimeout(() => {
+      this.setState({
+        isRegistered: false,
+      });
+    }, 3000);
+  }
+
+  allertMessage(message, variant = 'danger', dismissible = false) {
+    if (dismissible) {
+      return (
+        <Alert
+          variant={variant}
+          dismissible={dismissible}
+          onClose={() => { this.setState({ isBadRequestError: false }); }}
+        >
+          {message}
+        </Alert>
+      );
+    }
+    return (
+      <Alert variant={variant}>
+        {message}
+      </Alert>
+    );
+  }
+
+  backtoInfo() {
+    return this.props.history.push('/profile/info');
+  }
+
+  saveRestaurant() {
+    const restaurant = {
+      name: this.state.name,
+      email: this.state.email,
+      textAddress: this.state.address,
+      description: this.state.description,
+      workingTime: `${this.state.time_from}-${this.state.time_to}`,
+      personId: this.userID,
+      createdBy: this.userID,
+      modifyBy: this.userID,
+      tables: this.state.tables,
+    };
+
+    Api.post('restaurants', restaurant)
+      .then((response) => {
+        if (response.error) {
+          // eslint-disable-next-line no-console
+          console.error(response);
+          this.setState({
+            isBadRequestError: true,
+          });
+          return;
+        }
+        this.showRegistration();
+        this.props.history.push(`/restaurants/${response.data.id}`);
+      });
+  }
+
+  render() {
+    const { isRegistered } = this.state;
+    return (
+      <Container fluid className="restaurantRegistration">
+        <h4>
+          Restaurant registration
+        </h4>
+        <h6 className="warning">
+          Fields with (*) are required :
+        </h6>
+        <Form name="registrationForm">
+          {' '}
+          <div className="form_registration">
+            <div className="register_info_container">
+              <Form.Group controlId="name">
+                <Form.Label>Name *</Form.Label>
+                <Form.Control
+                  autoFocus
+                  name="name"
+                  value={this.state.name}
+                  onChange={this.handleChange}
+                />
+              </Form.Group>
+              <Form.Group controlId="email">
+                <Form.Label>Email *</Form.Label>
+                <Form.Control
+                  autoFocus
+                  name="email"
+                  type="email"
+                  value={this.state.email}
+                  onChange={this.handleChange}
+                />
+              </Form.Group>
+              {' '}
+              <Form.Group controlId="address">
+                <Form.Label>Address *</Form.Label>
+                <Form.Control
+                  autoFocus
+                  name="address"
+                  value={this.state.address}
+                  onChange={this.handleChange}
+                />
+              </Form.Group>
+              {' '}
+              <Form.Group controlId="workingTime">
```

Comment:
I can add wrong working time - on back-end we use simple `String` type without validation, but you add form with timepicker.